lib/api/src/grpc/conversions.rs

```diff
@@ -886,6 +892,29 @@ impl From<segment::types::GeoRadius> for GeoRadius {
     }
 }
 
+impl TryFrom<GeoPolygon> for segment::types::GeoPolygon {
+    type Error = Status;
+
+    fn try_from(value: GeoPolygon) -> Result<Self, Self::Error> {
+        if value.points.is_empty() {
+            return Err(Status::invalid_argument("Empty GeoPolygon"));
+        }
+
+        let points: Vec<segment::types::GeoPoint> =
+            value.points.into_iter().map(|point| point.into()).collect();
```

Comment:
`itertools` has nice sugar for this, you'll need to make sure it is imported
```suggestion
            value.points.into_iter().map_into().collect();
```

Otherwise we can also do this:
```suggestion
            value.points.into_iter().map(Into::into).collect();
```

Reply:
I prefer the latter so we don't need an extra import.

Reply:
done. Thanks! Learning some new syntax :)

Reply:
I believe  `clippy` usually recommends changes like this. FYI :smile: